core/translate/optimizer/cost.rs

```diff
@@ -39,27 +39,105 @@ pub fn estimate_page_io_cost(rowcount: f64) -> Cost {
 }
 
 /// Estimate the cost of a scan or seek operation.
-///
-/// This is a very simple model that estimates the number of pages read
-/// based on the number of rows read, ignoring any CPU costs.
 pub fn estimate_cost_for_scan_or_seek(
     index_info: Option<IndexInfo>,
     constraints: &[Constraint],
     usable_constraint_refs: &[RangeConstraintRef],
     input_cardinality: f64,
     base_row_count: f64,
 ) -> Cost {
+    // Check if this looks like real ANALYZE data (not the hardcoded fallback)
```

Comment:
Maybe `base_row_count` can be a 

```rust
enum RowCount {
  /// Hardcoded fallback default value in lieu of actual stats
  Fallback(f64),
  /// Statistics gathered via ANALYZE
  Statistic(f64)
}
```

or whatever, instead of this :]